Control/setting changes

mikechristie opened the pull request:
The following patches syncs up how controls and settings are handled for the gateway objects like the target, client and lun. This makes it easier to share and use the code with gwcli and ceph-ansible.

Commit:
Add helper to normalize a dict of controls

This will normalize a dict of controls. It will be used in gwcli to
convert between the user passed dict and json converted one and to also
verify input.

Signed-off-by: Mike Christie <mchristi@redhat.com>

Commit:
Use controls normalize helper to validate disk ops

The disk creation and reconfigure ops can pass in a dict of controls.
This uses the controls normalize function to verify them.

Signed-off-by: Mike Christie <mchristi@redhat.com>

dillaman reviewed (approved):
lgtm

dillaman commented on ceph_iscsi_config/lun.py:
```diff
@@ -364,18 +355,94 @@ def manage(self, desired_state):
+            host_group = self.config.config['groups'][group_name]
+            members = host_group.get('members')
+            disks = host_group.get('disks').keys()
+            if self.image_ not in disks:
```
Typo -- results in "'LUN' object has no attribute 'image_'" exception

Commit:
Make hw_max_sectors configurable

The tcmu hw_max_sectors is used to report the max/opt transfer sizes
to the initiator. The kernel def is only 128, so this increases our
default and also makes it configurable.

Signed-off-by: Mike Christie <mchristi@redhat.com>

Commit:
Handle tcmu control errors

In newer kernels writing invalid values to the control file will return
an error. Catch IOErrors instead of crashing.

Signed-off-by: Mike Christie <mchristi@redhat.com>

Commit:
mv control code to new class

This moves the control code to a new class so GWTarget, GWClient and
GWLUN ... ehhh ... LUN can all share it.

There are 2 interface changes.

1. The update_*_control functions return errors differently. The acl one
was passing the rtslib exception upwards but the target one was returning
a target.err. This patch syncs on the passing the rtslib exception upwards.

2. control reconfiguration was committed in the target manage function
but for the lun class we had a commit callout. This adds a generic
commit callback so it works the same for all classes.

Signed-off-by: Mike Christie <mchristi@redhat.com>

mikechristie commented:
Fixed image_ issue.

Also added new patch

https://github.com/ceph/ceph-iscsi-config/pull/77/commits/27612649014e55c6809542ae23bd20210e8346a2

to catch where tcmu, in recent kernels, returns an error if an invalid value is written to the tcmu/kernel control file.

Commit:
Move lun activate/deactivate from gwcli to config

The ansible code needs the same functionality when reconfiguring
settings so move activate/deactivate to from gwcli to ceph-iscsi-config.

Signed-off-by: Mike Christie <mchristi@redhat.com>

mikechristie commented:
Did a forced repush to fix issue where I was still using self.image instead of self.config_key in lun.activate. This resulted in the disk not getting remapped in the kernel.
